src/clarity-angular/emphasis/alert/alerts-pager.spec.ts

```diff
@@ -0,0 +1,125 @@
+*/
+import {AlertsPager} from "./alerts-pager";
+import {MultiAlertService} from "./providers/multi-alert-service";
+import {
+    Component,
+    QueryList,
+    Type,
+    ViewChild
+} from "@angular/core";
+import {TestBed} from "@angular/core/testing";
+import {Alert} from "./alert";
+import {ClrEmphasisModule} from "../../emphasis/emphasis.module";
+
+describe("Alerts pager component", function () {
+    describe("Typescript API", function () {
+        let component: AlertsPager;
+        let service: MultiAlertService;
+
+        beforeEach(() => {
+            service = new MultiAlertService();
+            component = new AlertsPager(service);
+        });
+
+        afterEach(() => {
+            service = null;
+            component = null;
+        });
+
+        it("page up calls the multi alert service", function () {
+            spyOn(service, "next").and.callFake(() => { });
+            component.pageUp();
+            expect(service.next).toHaveBeenCalled();
+        });
+
+        it("page down calls the multi alert service", function () {
+            spyOn(service, "previous").and.callFake(() => { });
+            component.pageDown();
+            expect(service.previous).toHaveBeenCalled();
+        });
+    });
+
+    describe("Template API", function () {
+
+        beforeEach(function () {
+            let service = new MultiAlertService();
+            let queryList = new QueryList<Alert>();
+
+            this.create = <T>(componentType: Type<T>) => {
+                TestBed.configureTestingModule({
+                    imports: [ClrEmphasisModule],
+                    declarations: [componentType],
+                    providers: [
+                        {
+                            provide: MultiAlertService,
+                            useValue: service
+                        }
+                    ]
+                });
+                this.fixture = TestBed.createComponent(componentType);
+
+                this.alertFixture = TestBed.createComponent(Alert);
+                this.alert = this.alertFixture.componentInstance;
+                this.secondAlertFixture = TestBed.createComponent(Alert);
+                this.secondAlert = this.secondAlertFixture.componentInstance;
+
+                queryList.reset([
+                    this.alert,
+                    this.secondAlert
+                ]);
+                service.manage(queryList);
+
+                this.fixture.detectChanges();
+            };
+        });
+
+        afterEach(function() {
+            this.fixture.destroy();
+            this.alertFixture.destroy();
+            this.secondAlertFixture.destroy();
+        });
+
+        it("offers two way binding on the alert index", function () {
+            this.create(TestIndex);
+            this.fixture.componentInstance.index = 0;
+            this.fixture.detectChanges();
+            expect(this.fixture.componentInstance.pagerInstance.currentAlertIndex).toEqual(0);
```

Comment:
Testing with the default value is a bit of a risk, if it didn't work you wouldn't know. It's probably safer to use 1 to check the input and then revert to 0 to check the output, rather than the opposite.

Reply:
Reversed it.